rootex/core/renderer/post_processor.cpp

```diff
@@ -153,6 +153,55 @@ class MonochromePostProcess : public PostProcess
 	}
 };
 
+Variant CustomPostProcess::loadRTVAndSRV(const Event* event)
+{
+	RenderingDevice::GetSingleton()->createRTVAndSRV(m_CacheRTV, m_CacheSRV);
+	return true;
+}
+
+CustomPostProcess::CustomPostProcess(String path)
+{
+	postProcessPath = path;
+
+	m_Binder.bind(RootexEvents::WindowResized, this, &CustomPostProcess::loadRTVAndSRV);
+
+	loadRTVAndSRV(nullptr);
+
+	m_BasicPostProcess.reset(new DirectX::BasicPostProcess(RenderingDevice::GetSingleton()->getDevice()));
+}
+
+void CustomPostProcess::draw(CameraComponent* camera, ID3D11ShaderResourceView*& nextSource)
+{
+	const PostProcessingDetails& postProcessingDetails = camera->getPostProcessingDetails();
+	for (auto&& details : postProcessingDetails.m_CustomPostProcessing)
+	{
+		if (details.path == postProcessPath && details.isSelected == true)
```

Comment:
is `details.path == postProcessPath` necessary? this function would only be called if this object is created and added to the camera in the first place, in which case we only need to check `details.isSelected == true`. Also doesn't make sense to keep a loop here, you can change the prototype of `draw` to pass `isSelected` from the caller itself.

We need to minimize string comparisons in such methods that are called per frame